src/db.ts

```diff
@@ -77,31 +84,34 @@ export class NostroidDexie extends Dexie {
       })
     })
   }
-  
-  public async nip05Valid(name, pubkey) {
-    if (name == undefined || name.length < 3) {
+  private async nip05Valid(name: string, pubkey: string) {
+    if (typeof name !== 'string' || name.length < 3) {
       return false
     }
-    var n = name
+    let n = name
     switch (name.indexOf('@')) {
       case -1: n = `_@${name}`; break
       case 0: n = `_${name}`; break
       default: n = name
     }
     const qName = await queryName(n)
     console.log(`${name}->${n}->${qName}`)
-    return qName == pubkey
+    return qName === pubkey
   }
-  
-  public async updateProfileFromMeta(pubkeys) {
+
+  public async updateProfileFromMeta(pubkeys: string[]) {
     console.log(`updateProfileFromMeta(${pubkeys?.length})`)
-    if (pubkeys && pubkeys.length === 0) {
+    if (Array.isArray(pubkeys) && pubkeys.length === 0) {
       return
     }
-    let profiles: Array<IProfile>
-    await db.transaction('rw', db.profiles, db.events, async () => {
+
+    const profiles = await db.transaction('rw', db.profiles, db.events, async () => {
+      let profiles: Array<IProfile> = []
+
       if (pubkeys) {
-        profiles = await db.profiles.where('pubkey').anyOf(pubkeys).toArray()
+        profiles = (
+          await Promise.all(
+            pubkeys.map((key) => db.profiles.where('pubkey').equals(key).toArray()))).flat();
```

Comment:
Dexie/IDB is optimized to search for `where('pubkey').anyOf(pubkeys)` while you shoot at once multiple queries at the DB here. Unless this is a weird hack that actually works, please keep it one query using `anyOf()`.

Also looking for a value by its key should not need `.toArray()` but `.first()`.

Reply:
Added comments to explain this, but the creator of Dexie recommends this instead: https://github.com/dexie/Dexie.js/issues/1388#issuecomment-917367428

As does a commenter on the same issue: https://github.com/dexie/Dexie.js/issues/1388#issuecomment-918898007

The average time on my deployment (https://nostroid-test.vercel.app/) has been considerably more consistent and is lower (on my machine anyways) than the build deployed on https://chat.nostr.info/

Not to mention that a quick look at the Dexie source code, it appears that `anyOf` is still using a normal IDBCursor. 

https://github.com/dexie/Dexie.js/blob/420e39632c4298b41ce61c53b4edf815be785f61/src/classes/where-clause/where-clause.ts#L179

https://github.com/dexie/Dexie.js/blob/fe682ef24568278c3b31d9d6c93de095d4b77ae8/src/dbcore/dbcore-indexeddb.ts#L193

They return the type `DBCoreCursor` because they wrap the IDBCursor in a proxy to make it more usable.

Reply:
So `anyOf` indeed does not use the index! Holy cow! This is certainly very valuable to know but here in this case potentially not worth it as the profiles table doesn't grow that big. Or so I thought. I did the test, comparing:

```
        console.time('get profiles');
        profiles = await db.profiles.where('pubkey').anyOf(pubkeys).toArray();
        console.timeEnd('get profiles');
```

vs.

```
        console.time('get profiles');
        profiles = (
          await Promise.all(
            pubkeys.map((key) => db.profiles.where('pubkey').equals(key).first())));
        console.time('get profiles');
```

and got this:

```
     snuffyDev	      giszmo
           0.7          1.6
           0.8          1.2
           2.8          0.7
           1.3          0.7
           4.7      1,345.8
          34.2      1,292.2
           0.4      2,513.8
           0.6          1.3
           0.5          5.5
           0.4         17.3
           0.9        552.0
           0.4        274.9
           2.0          3.3
           0.6	
          20.5	
         147.8	
          61.7	
           0.8	
min        0.4          0.7
median     0.9          5.5
average   15.6        462.3
max      147.8      2,513.8
standard dev 36.7     785.4
```